src/main/webapp/consentGroup/NewConsentGroup.js

```diff
@@ -71,10 +73,14 @@ class NewConsentGroup extends Component {
   }
 
   componentDidMount() {
+    this._isMounted = true;
     this.initDocuments();
     this.getUserSession();
-    ConsentGroup.getConsentGroupNames().then(
-      resp => this.setState({ existingGroupNames: resp.data }));
+    ConsentGroup.getConsentGroupNames().then(resp => {
+      if (this._isMounted) {
```

Comment:
I'm not sure why we are overriding how React handles component mounting. This line is always true since it is set to true on line 76. React will call this method right after the component mounts, so we should not have to manually track the lifecycle with `_isMounted`.

Reply:
Actually it's not always true. Since this is a promise this will be send to the queue of the jvm so when the component unmounts the var `_isMounted` is setted to false and because is the callback of the promise it keeps the reference inside the promise. When the promise resolves it will have `_isMounted` properly set.

Reply:
Gotcha, yes, makes sense. Also addressed in slack.